lib/routes/kanxue/topic.ts

```diff
@@ -0,0 +1,148 @@
+    | Pwn          | pwn        |
+    | WEB 安全     | web        |
+    | 极客空间   | geekzone |
+    | 外文翻译     | translate  |
+    | 全站         | all        |
+    
+    | 类型     | type   |
+    | -------- | ------ |
+    | 最新主题 | latest |
+    | 精华主题 | digest |`,
+};
+
+async function handler(ctx) {
+    const category = ctx.req.param('category') || 'all';
+    const type = ctx.req.param('type') || 'latest';
+    let path;
+    let title;
+    if (Object.hasOwn(categoryId, category)) {
+        if (type === 'digest') {
+            // type为digest时只获取精华帖
+            path = `forum-${categoryId[category][0]}-1.htm?digest=1`;
+            title = `看雪论坛精华主题 - ${categoryId[category][1]}`;
+        } else {
+            // type为空/非法/latest时则只获取最新帖
+            path = `forum-${categoryId[category][0]}.html`;
+            title = `看雪论坛最新主题 - ${categoryId[category][1]}`;
+        }
+    } else {
+        // category未知时则获取全站最新帖
+        if (category === 'digest') {
+            path = 'new-digest.htm';
+            title = '看雪论坛精华主题';
+        } else {
+            path = 'new-tid.htm';
+            title = '看雪论坛最新主题';
+        }
+    }
+
+    const response = await got({
+        method: 'get',
+        url: baseUrl + path,
+        headers: {
+            Referer: baseUrl,
+        },
+    });
+
+    const $ = load(response.data);
+    const list = $('.thread');
+
+    const resultItem = await Promise.all(
+        list
+            ? list
+                // fix .thread .top_3
+                .filter((_, elem) => {
+                    const pubDate = kanxueUtils.dateParser($('.date', elem).eq(0).text(), 8);
+                    return !elem.attribs.class.includes('top') || Date.now() - new Date(pubDate) < 1000 * 60 * 60 * 24 * 3;
+                })
+                .map((_, elem) => {
+                    const subject = $('.subject a', elem).eq(1);
+                    const pubDate = kanxueUtils.dateParser($('.date', elem).eq(0).text(), 8);
+
+                    const link = `${baseUrl}${subject.attr('href')}`;
+                    const key = `kanxue: ${link}`;
+
+                    return cache.tryGet(key, async () => {
+                        const postDetail = await got({
+                            method: 'get',
+                            url: link,
+                        });
+                        const $ = load(postDetail.data);
+                        $('.card')
+                            .eq(0)
+                            .find('.message img')
+                            .each((_, item) => {
+                                item = $(item);
+
+                                const src = item.attr('src');
+                                if (src !== undefined && !src.startsWith('https://') && !src.startsWith('http://')) {
+                                    item.attr('src', `https://bbs.kanxue.com/${src}`);
+                                }
+                                item.attr('referrerpolicy', 'no-referrer');
```

Comment:
RSSHub will set referer policy in middleware.
https://github.com/DIYgod/RSSHub/blob/3ba4a634cf83f2217e2722832216d71c5d33e89b/lib/middleware/parameter.ts#L153-L155